apps/gdalalg_raster_tile.cpp

```diff
@@ -2894,9 +3630,66 @@ bool GDALRasterTileAlgorithm::RunImpl(GDALProgressFunc pfnProgress,
 
     CPLWorkerThreadPool oThreadPool;
 
+    bool bThreadPoolInitialized = false;
+    const auto InitThreadPool =
+        [this, nBaseTiles, &oThreadPool, &bRet, &bThreadPoolInitialized]()
     {
+        if (!bThreadPoolInitialized)
+        {
+            bThreadPoolInitialized = true;
+            m_numThreads = std::max(1, m_numThreads);
+            if (static_cast<uint64_t>(m_numThreads) > nBaseTiles)
+                m_numThreads = static_cast<int>(nBaseTiles);
+
+            if (bRet && m_numThreads > 1)
+            {
+                CPLDebug("gdal_raster_tile", "Using %d threads", m_numThreads);
+                bRet = oThreadPool.Setup(m_numThreads, nullptr, nullptr);
+            }
+        }
+
+        return bRet;
+    };
+
+    // Just for unit test purposes
+    const bool bEmitSpuriousCharsOnStdout = CPLTestBool(
+        CPLGetConfigOption("GDAL_RASTER_TILE_EMIT_SPURIOUS_CHARS", "NO"));
+
+    constexpr int THRESHOLD_TILES_PER_JOB = 100;
+    const auto IsCompatibleOfSpawnSilent = [this]()
+    {
+        CPLErrorStateBackuper oBackuper(CPLQuietErrorHandler);
+        const char *pszErrorMsg = "";
+        return IsCompatibleOfSpawn(pszErrorMsg);
+    };
+
+    if (m_ovrZoomLevel >= 0)
+    {
+        // do not generate base tiles if called as a child process with
+        // --ovr-zoom-level
+    }
+    else if (m_numThreads > 1 &&
+             ((m_parallelMethod.empty() &&
+               nBaseTiles / m_numThreads > THRESHOLD_TILES_PER_JOB &&
```

Comment:
This is the wrong criterion for switching from threads to spawning processes. The degree of parallelism numThreads should be limited to ceil(nXXXTiles / THRESHOLD_TILES_PER_JOB) and spawning used if numThreads > 8 or so. With the current method, if I ask for -j 128 but there are only 12,000 tiles I'll get 128 threads (at least that's what it looks like).